extension/src/popup/components/account/AccountAssets/index.tsx

```diff
@@ -47,6 +47,10 @@ export const AccountAssets = ({
   const networkDetails = useSelector(settingsNetworkDetailsSelector);
   const [hasIconFetchRetried, setHasIconFetchRetried] = useState(false);
 
+  useEffect(() => {
+    setAssetIcons(inputAssetIcons);
+  }, [inputAssetIcons]);
+
```

Comment:
this was why USDC icons (for ex) weren't showing up. We needed to reset the state when the passed prop updates

Reply:
ahh weird, I would think it would get re-rendered when redux loads the assetIcons? 🤔  but nice fix!

Reply:
Yeah, I thought the same, too. It appears that once `useState`'s default state is set, it doesn't update again until you manually set it. I guess it makes it more deterministic and prevents too many re-renders